common/djangoapps/util/password_policy_validators.py

```diff
@@ -1,322 +1,467 @@
+from django.contrib.auth.password_validation import (
+    get_default_password_validators,
+    CommonPasswordValidator as DjangoCommonPasswordValidator,
+    MinimumLengthValidator as DjangoMinimumLengthValidator,
+    UserAttributeSimilarityValidator as DjangoUserAttributeSimilarityValidator,
+)
 from django.core.exceptions import ValidationError
-from django.utils.translation import ugettext_lazy as _
-from django.utils.translation import ungettext_lazy as ungettext
-from Levenshtein import distance
+from django.utils.translation import ugettext as _, ungettext
 from six import text_type
 
 from student.models import PasswordHistory
 
-
 log = logging.getLogger(__name__)
 
-# In description order
-_allowed_password_complexity = [
-    'ALPHABETIC',
-    'UPPER',
-    'LOWER',
-    'NUMERIC',
-    'DIGITS',
-    'PUNCTUATION',
-    'NON ASCII',
-    'WORDS',
-]
 
-
-class SecurityPolicyError(ValidationError):
-    pass
+def password_validators_instruction_texts(password_validators=None):
+    """
+    Return a string of instruction texts of all configured validators.
+    Expects at least the MinimumLengthValidator to be defined.
+    """
+    complexity_instructions = []
+    # For clarity in the printed instructions, the minimum length instruction
+    # is separated from the complexity instructions.
+    length_instruction = ''
+    if password_validators is None:
+        password_validators = get_default_password_validators()
+    for validator in password_validators:
+        text = validator.get_instruction_text()
+        if text:
+            if isinstance(validator, MinimumLengthValidator):
+                length_instruction = text
+            else:
+                complexity_instructions.append(text)
+    if complexity_instructions:
+        return _('Your password must contain {length_instruction}, including {complexity_instructions}.').format(
+            length_instruction=length_instruction,
+            complexity_instructions=' & '.join(complexity_instructions)
+        )
+    else:
+        return _('Your password must contain {length_instruction}.'.format(length_instruction=length_instruction))
 
 
-def password_min_length():
+def password_validators_restrictions(password_validators=None):
     """
-    Returns minimum required length of a password.
-    Can be overridden by site configuration of PASSWORD_MIN_LENGTH.
+    Return a dictionary of complexity restrictions to be used by mobile users on
+    the registration form
     """
-    min_length = getattr(settings, 'PASSWORD_MIN_LENGTH', None)
-    if min_length is None:
-        return 2  # Note: This default is simply historical
-    return min_length
+    complexity_restrictions = {}
+    if password_validators is None:
+        password_validators = get_default_password_validators()
+    for validator in password_validators:
+        key, value = validator.get_restriction()
+        if key:
+            complexity_restrictions[key] = value
```

Comment:
Secondly, we require restrictions? Maybe for our special add-ons like instruction/restrictions, we could allow the lack of the member function. So that OpenEdx people can more easily integrate validators that we didn't write (or even so we can use UserAttributeSimilarityValidator as is, without writing an empty wrapper).

So maybe something like:
`complexity = dict(validator.get_restriction() for validator in password_validators if hasattr(validator, 'get_restriction'))`